### PR TITLE
improvement: Add option to watch unmanaged inputs

### PR DESCRIPTION
**Pull request by tgodzik**
This allows us to rerun source generators if the scripts change.

Fixes https://github.com/scalacenter/bloop/issues/2386

### CODE REVIEW

File: frontend/src/main/scala/bloop/engine/SourceGenerator.scala [hunk 1]
```diff
@@ -20,6 +20,7 @@ final case class SourceGenerator(
     cwd: AbsolutePath,
     sourcesGlobs: List[SourcesGlobs],
     outputDirectory: AbsolutePath,
+    unamangedInputs: List[AbsolutePath],
```
Comment (bishabosha) [inline frontend/src/main/scala/bloop/engine/SourceGenerator.scala:23]:
```suggestion
    unmanagedInputs: List[AbsolutePath],
```

File: frontend/src/main/scala/bloop/engine/SourceGenerator.scala [hunk 2]
```diff
@@ -61,6 +62,7 @@ final case class SourceGenerator(
 
   private def run(
       inputs: Map[AbsolutePath, Int],
+      unamangedInputs: Map[AbsolutePath, Int],
```
Comment (bishabosha) [inline frontend/src/main/scala/bloop/engine/SourceGenerator.scala:65]:
```suggestion
      unmanagedInputs: Map[AbsolutePath, Int],
```

**Review by bishabosha — APPROVED** [review]

**Comment by tgodzik** [comment]
Scheduled for 2.0.1

**Comment by bishabosha** [comment]
@Perklone relevant for you

**Pull request merged**